### PR TITLE
Add documentation for `ExitStatus`

### PR DESCRIPTION
**Pull request by citizen428**
As requested in #29370. r? @steveklabnik

### CODE REVIEW

File: src/libstd/process.rs [hunk 1]
```diff
@@ -754,6 +754,11 @@ impl fmt::Debug for Stdio {
 }
 
 /// Describes the result of a process after it has terminated.
+///
+/// This `struct` is used to represent the exit status of a child process.
+/// Child processes are created via the [`Command`] struct and their exit
+/// status is exposed through the [`status`] method.
```
Comment (Mark-Simulacrum) [inline src/libstd/process.rs]: I think these two (`status` and `Command`) are missing the URL portion.

Comment (steveklabnik) [inline src/libstd/process.rs]: Yup!

Comment (citizen428) [inline src/libstd/process.rs]: @steveklabnik @Mark-Simulacrum I added them just now.

File: src/libstd/process.rs [hunk 2]
```diff
@@ -788,6 +795,22 @@ impl ExitStatus {
     /// On Unix, this will return `None` if the process was terminated
     /// by a signal; `std::os::unix` provides an extension trait for
     /// extracting the signal and other details from the `ExitStatus`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
```
Comment (GuillaumeGomez) [inline src/libstd/process.rs]: The rust tag is useless (it's enabled by default).

Comment (citizen428) [inline src/libstd/process.rs]: Removed it. I had actually copied this over from another example in the same file.

**Comment by steveklabnik** [comment]
Yeah, no worries; some of the stuff is in an older style.

File: src/libstd/process.rs [hunk 3]
```diff
@@ -788,6 +795,22 @@ impl ExitStatus {
...
+    /// let status = Command::new("mkdir")
+    ///                      .arg("projects")
+    ///                      .status()
+    ///                      .expect("failed to execute mkdir");
+    ///
+    /// match status.code() {
+    ///     Some(code) => println!("Exited with status code: {}", code),
+    ///     None       => println!("Process terminated by signal")
+    /// }
+    /// ...
```
Comment (ollie27) [inline src/libstd/process.rs]: `...` → ```` ``` ````

**Comment by citizen428** [comment]
Needing several follow-up commits for a doc PR, not my proudest moment 😂

**Comment by steveklabnik** [comment]
It's no big deal at all, it happens all the time :)

**Comment by GuillaumeGomez** [comment]
Two things now: tests seem to fail, so please fix them. :p Then please squash your commits.

**Comment by citizen428** [comment]
@GuillaumeGomez Yeah, I was about to squash the commits. Re the failing specs, I was wondering how a docs only commit would break anything, gotta look into that when I have some time.

**Comment by steveklabnik** [comment]
> I was wondering how a docs only commit would break anything

Code examples are run as tests!

**Comment by GuillaumeGomez** [comment]
In your case this is an invalid linkage:

```
std/process/struct.ExitStatus.html:56: broken link fragment `#method.status` pointing to `std/process/struct.ExitStatus.html`
```

**Comment by citizen428** [comment]
I see, doctests. Well, this commit hopefully will finally fix this issue.

**Comment by steveklabnik** [comment]
It did! 🎊 

@bors: r+ rollup

**Comment by alexcrichton** [comment]
@bors: retry

* [appveyor network issues](https://appveyor.statuspage.io/incidents/2g5jnq5cy1g1) #40474

**Pull request merged**